### PR TITLE
Add test for LabelWebhookController

### PR DESCRIPTION
**Pull request by ThePnx**

### CODE REVIEW

**Review by Vortim — DISMISSED** [review]

File: src/test/resources/github/label/exampleLabelPayload.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (mateuszmarek) [inline src/test/resources/github/label/exampleLabelPayload.json]: I would rename the file to `exampleLabelWebhookRequest.json` as `payload` does not reflect the fact that it's a request from github's webhook ;)

File: src/test/java/rotateit/controller/LabelWebhookControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.standaloneSetup;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration
+@WebAppConfiguration
+public class LabelWebhookControllerTest {
+    private MockMvc helper;
```
Comment (mateuszmarek) [inline src/test/java/rotateit/controller/LabelWebhookControllerTest.java]: I believe convention is to simply name it `mockMvc`, also `helper` doesn't tell much

File: src/test/java/rotateit/controller/LabelWebhookControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    @InjectMocks
+    private LabelWebhookController labelWebhookController;
```
Comment (mateuszmarek) [inline src/test/java/rotateit/controller/LabelWebhookControllerTest.java]: from what i see there are no mocks to inject, why not simply `@Autowired`? Also, you're using `SpringJUnit4ClassRunner` and i don't see any `initMocks()`, should this even work ?

Comment (ThePnx) [reply src/test/java/rotateit/controller/LabelWebhookControllerTest.java]: Leftover... 🤔

Comment (ThePnx) [inline src/test/java/rotateit/controller/LabelWebhookControllerTest.java]: Actually it have to be annotated, for now we don't have context.

Comment (mateuszmarek) [reply src/test/java/rotateit/controller/LabelWebhookControllerTest.java]: Ok, do it like this:
You don't need context, create controller object in setup and remove `@InjectMocks`
`labelWebhookController = new LabelWebhookController();`
You also don't need `@ContextConfiguration` so:
```
@RunWith(SpringJUnit4ClassRunner.class)
@WebAppConfiguration
public class LabelWebhookControllerTest {
    private MockMvc helper;

    private LabelWebhookController labelWebhookController;

    @Before
    public void setUp() {
        labelWebhookController = new LabelWebhookController();
        helper = standaloneSetup(labelWebhookController).build();
    }
```

File: src/test/java/rotateit/controller/LabelWebhookControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.standaloneSetup;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration
+@WebAppConfiguration
+public class LabelWebhookControllerTest {
+    private MockMvc mockMvc;
+
+    private LabelWebhookController labelWebhookController = mock(LabelWebhookController.class);
```
Comment (mateuszmarek) [inline src/test/java/rotateit/controller/LabelWebhookControllerTest.java]: This is integration test, don't mock the controller you're testing!

File: src/test/java/rotateit/controller/LabelWebhookControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+import static com.google.common.base.Charsets.UTF_8;
+import static org.mockito.Mockito.mock;
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.standaloneSetup;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration
```
Comment (mateuszmarek) [inline src/test/java/rotateit/controller/LabelWebhookControllerTest.java]: Why didn't you remove `@ContextConfiguration` ?

File: src/test/resources/github/label/exampleLabelWebhookRequest.json [hunk 6]
```diff
@@ -0,0 +1,24 @@
+{
+  "action": "labeled",
+  "number": 1,
+  "pull_request": {
+    "id": 126624344,
+    "html_url": "https://github.com/dkierecki/e-rotator/pull/1",
```
Comment (dkierecki) [inline src/test/resources/github/label/exampleLabelWebhookRequest.json]: Please use fake data.

File: src/test/resources/github/label/exampleLabelWebhookRequest.json [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    "number": 1,
+    "state": "open",
+    "title": "Update .gitignore",
+    "user": {
+      "login": "dkierecki",
```
Comment (dkierecki) [inline src/test/resources/github/label/exampleLabelWebhookRequest.json]: Please use fake data.

File: src/test/resources/github/label/exampleLabelWebhookRequest.json [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+      "id": 11491530,
+      "type": "User",
+      "site_admin": false
+    }
+  },
+  "label": {
+    "id": 629619391,
+    "url": "https://api.github.com/repos/dkierecki/e-rotator/labels/bug",
```
Comment (dkierecki) [inline src/test/resources/github/label/exampleLabelWebhookRequest.json]: Please use fake data.

File: src/test/resources/github/label/exampleLabelWebhookRequest.json (hunk 7 above)
Comment (mateuszmarek) [reply src/test/resources/github/label/exampleLabelWebhookRequest.json]: :D

**Review by dkierecki — APPROVED** [review]

**Pull request merged**